util/promlint/promlint.go

```diff
@@ -205,6 +210,54 @@ func lintHistogramSummaryReserved(mf dto.MetricFamily) []Problem {
 	return problems
 }
 
+// lintMetricTypeInName detects when the type of the metric is included in the metric name
+func lintMetricTypeInName(mf dto.MetricFamily) []Problem {
+	t := mf.GetType()
+	if t == dto.MetricType_UNTYPED {
+		return nil
+	}
+
+	var problems problems
+	n := strings.ToLower(mf.GetName())
+
+	typename := strings.ToLower(dto.MetricType_name[int32(t)])
```

Comment:
`dto.MetricType_name[int32(t)]` could be simplified to `t.String()`

Reply:
Fixed.